### PR TITLE
"replaceAll" modified to replace in Criteria.class

### PR DESCRIPTION
**Pull request by SinghAdes**
As per my understanding,  Line:-117 & 125  "field.get(0).replaceAll("^'|'$", "")" is used for removing single quotes from the field. That can be done with "field.get(0).replace("'", "")" also.

### CODE REVIEW

**Comment by SinghAdes** [comment]
@julianladisch, Raising separate PR for similar issues

File: domain-models-runtime/src/main/java/org/folio/rest/persist/Criteria/Criteria.java [hunk 1]
```diff
@@ -114,15 +114,15 @@ private void createFromSnippet() {
       from = new From();
       from.setSnippet(ARRAY_FROM_CLAUSE + "("
         + PostgresClient.DEFAULT_JSONB_FIELD_NAME + GET_JSON_FIELD + arrayField + ")");
-      from.setAsValue(field.get(0).replaceAll("^'|'$", "")); //remove ''
+      from.setAsValue(field.get(0).replace("'", "")); //remove ''
     }
```
Comment (adamdickmeiss) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/Criteria/Criteria.java:118]: Not the same .. Eg the original would not remove quote from `a'b` , but the new one will. I believe all quotes should be removed, so OK by me.

**Review by adamdickmeiss — APPROVED** [review]
OK by me (even with the change of behavior).

File: domain-models-runtime/src/main/java/org/folio/rest/persist/Criteria/Criteria.java (hunk 1 above)
Comment (SinghAdes) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/Criteria/Criteria.java:118]: Ok thanks.

File: domain-models-runtime/src/main/java/org/folio/rest/persist/facets/FacetField.java [hunk 2]
```diff
@@ -23,6 +23,7 @@ public FacetField(String path2facet, int amountOfValues2return){
     this(path2facet, amountOfValues2return, null);
   }
 
+  @SuppressWarnings("java:S5850")//Line:34 (in regex '$' has higher precedence than '|' , but it is a small regex so we can keep it)
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/facets/FacetField.java:26]: I would not suppress the warning but drop the first regexp and simply use the QUOTES_PATTERN regexp for both:

```
private static final Pattern QUOTES_PATTERN = Pattern.compile("'([^\']*)'");

  private FacetField(String path2facet, int topFacets2return, String alias){
    this.fieldPath = path2facet;
    this.topFacets2return = topFacets2return;
    if (path2facet == null || !path2facet.contains("\'")) {
      throw new ExceptionInInitializerError("Path to facet must be in the form of jsonb_field_name->'field1'->>'field2', for example jsonb->>'field'");
    }
    // get last occurrence of what is in between ''
    Matcher m = QUOTES_PATTERN.matcher(path2facet);
    while (m.find()) {
      this.alias = m.group(1);
    }
  }
```

File: domain-models-runtime/src/main/java/org/folio/rest/tools/client/HTTPJsonResponseHandler.java [hunk 3]
```diff
@@ -64,6 +64,11 @@ public void handle(AsyncResult<HttpResponse<Buffer>> res) {
         HttpModuleClient2.cache.put(endpoint, cf.get());
       } catch (Exception e) {
         log.error(e.getMessage(), e);
+
+        //InterruptedExceptions should never be ignored in the code
+        if(e instanceof InterruptedException){
+        //Restore interrupted state...
+         Thread.currentThread().interrupt();}
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/tools/client/HTTPJsonResponseHandler.java:71]:
```suggestion
        // InterruptedExceptions should never be ignored in the code
        if (e instanceof InterruptedException) {
          // Restore interrupted state
          Thread.currentThread().interrupt();
        }
```

**Pull request merged**